node/sc/bridge_manager.go

```diff
@@ -203,41 +192,37 @@ func (bi *BridgeInfo) GetCounterPartToken(token common.Address) common.Address {
 	return cpToken
 }
 
-func (bi *BridgeInfo) GetPendingRequestEvents() []*RequestValueTransferEvent {
+func (bi *BridgeInfo) GetPendingRequestEvents() []IRequestValueTransferEvent {
 	ready := bi.pendingRequestEvent.Pop(maxPendingNonceDiff / 2)
-	var readyEvent []*RequestValueTransferEvent
-	for _, item := range ready {
-		readyEvent = append(readyEvent, item.(*RequestValueTransferEvent))
+	readyEvent := make([]IRequestValueTransferEvent, len(ready))
+	for i, item := range ready {
+		readyEvent[i] = item.(IRequestValueTransferEvent)
 	}
-
 	vtPendingRequestEventCounter.Dec((int64)(len(ready)))
-
 	return readyEvent
 }
 
 // processingPendingRequestEvents handles pending request value transfer events of the bridge.
-func (bi *BridgeInfo) processingPendingRequestEvents() error {
+func (bi *BridgeInfo) processingPendingRequestEvents() {
 	ReadyEvent := bi.GetReadyRequestValueTransferEvents()
 	if ReadyEvent == nil {
-		return nil
+		return
 	}
 
 	logger.Trace("Get ready request value transfer event", "len(readyEvent)", len(ReadyEvent), "len(pendingEvent)", bi.pendingRequestEvent.Len())
 
 	for idx, ev := range ReadyEvent {
-		if ev.RequestNonce < bi.lowerHandleNonce || bi.handledEvent.Exist(ev.RequestNonce) {
-			logger.Trace("handled requests can be ignored", "RequestNonce", ev.RequestNonce, "lowerHandleNonce", bi.lowerHandleNonce)
+		if ev.GetRequestNonce() < bi.lowerHandleNonce || bi.handledEvent.Exist(ev.GetRequestNonce()) {
+			logger.Trace("handled requests can be ignored", "RequestNonce", ev.GetRequestNonce(), "lowerHandleNonce", bi.lowerHandleNonce)
 			continue
 		}
 
 		if err := bi.handleRequestValueTransferEvent(ev); err != nil {
 			bi.AddRequestValueTransferEvents(ReadyEvent[idx:])
 			logger.Debug("Failed handle request value transfer event", "err", err, "len(RePutEvent)", len(ReadyEvent[idx:]))
-			return err
+			return
```

Comment:
Does an error occur frequently or normally? If not, we better increase the log level from debug to warn because it is the last point the error is handled

Reply:
Rarely happen. How about changing it to `Error`?

Reply:
@hyunsooda Error is a possible option. If it makes some failures of value transfer requests permanently, I think Error is better. If we have another retrier of other back-up plans, Warn is okay for that.

Reply:
By call this `bi.AddRequestValueTransferEvents(ReadyEvent[idx:]), the event failed gets have a chance to be handled again. However, it is not always guaranteed to be successfully handled because of some reasons such as invalid nonce. It results in stacks of the pending transactions forever. Please refer to the fix in #1295. Logically, this is a failed behavior. So, I think `Error` seems reasonable.